bodhi/tests/functional/test_updates.py

```diff
@@ -2520,6 +2520,39 @@ def test_edit_testing_update_reset_karma_with_same_tester(self, publish, *args):
 
     @mock.patch(**mock_valid_requirements)
     @mock.patch('bodhi.notifications.publish')
+    def test_disable_autopush_for_critical_updates(self, publish, *args):
+        """
+        Make sure that autopush is disabled if critical update
+        receives any negative karma
+        """
+        user = User(name=u'bob')
+        self.db.add(user)
+        self.db.flush()
+
+        nvr = u'kernel-3.11.5-300.fc17'
+        args = self.get_update(nvr)
+        args['autokarma'] = True
+        args['stable_karma'] = 2
+        args['unstable_karma'] = -2
+        resp = self.app.post_json('/updates/', args)
+        self.assertTrue(resp.json['critpath'])
+        self.assertEquals(resp.json['request'], 'testing')
+        publish.assert_called_with(topic='update.request.testing', msg=ANY)
+
+        upd = self.db.query(Update).filter_by(title=resp.json['title']).one()
+        upd.status = UpdateStatus.testing
+        upd.request = None
+
+        upd.comment(self.db, u'Failed to work', author=u'ralph', karma=-1)
```

Comment:
I believe this test might not be hitting the new code path. The code has an if statement to check if the karma has reached the autopush threshold before the new code will execute. I could be misreading the conditions here, but it seems to me that the karma on ```upd``` is -1 and thus wouldn't push anyway. I think you probably need three upvoters after this -1 to reach an overall karma of +2 before this test will hit the new code path that is introduced in this commit. Does that make sense?

I didn't actually run this test through pdb to be sure, so I could be incorrect and I apologize if I am ☺